specs/string_reverse_spec.rb

```diff
@@ -2,53 +2,70 @@
 require 'minitest/reporters'
 require_relative '../lib/string_reverse'
 
-describe "basic test 1 - string reverse" do
-  it "reverse a string with odd number of characters" do
-    test_string = "Hello"
-    string_reverse(test_string)
-    test_string.must_equal "olleH"
-  end
-end
+describe "string reverse" do
+  describe "basic tests" do
+    describe "test 1" do
+      it "reverse a string with odd number of characters" do
+        test_string = "Hello"
 
-describe "basic test 2 - string reverse" do
-  it "reverse a string with even number of characters" do
-    test_string = "Software"
-    string_reverse(test_string)
-    test_string.must_equal "erawtfoS"
-  end
-end
+        string_reverse(test_string)
+
+        test_string.must_equal "olleH"
+      end
+    end
+    describe "test 2" do

```

Comment:
I think you're using extra `describe` blocks here. Since this is really modeled off of a BDD framework, ideally everything that is a text block has some sort of explanation for the expected behavior. I think you could have the single `describe` block at the top-level for "basic tests" and then dive straight into the `it` blocks that have the appropriate behavior descriptions. (The "test 1" and "test 2" describe blocks are really what is concerning, so you could either add additional descriptions to these or collapse them down - whichever makes sense)

Reply:
Sounds good! It's likely that I misunderstood Charles's feedback and am glad that you are reviewing it. Thank you.